---
title: "SignatureSets: An R Package for RNA-Based Immuno-Oncology Signatures"
author:
- name: Farnoosh Abbas-Aghababazadeh
  affiliation:
  - &pm Bioinformatics and Computational Genomics Laboratory, Princess Margaret Cancer Center,University Health Network, Toronto, Ontario, Canada
- name: Yacine Bareche
  affiliation:
  - &pm Faculty of Pharmacy, Université de Montréal, Montreal, Canada
  - &mbp Centre de Recherche du Centre Hospitalier de l’Université de Montréal, Institut du Cancer de Montréal, Montreal, Canada
- name: John Stagg
  affiliation:
  - &pm Faculty of Pharmacy, Université de Montréal, Montreal, Canada
  - &mbp Centre de Recherche du Centre Hospitalier de l’Université de Montréal, Institut du Cancer de Montréal, Montreal, Canada
- name: Benjamin Haibe-Kains
  affiliation:
    - &pm Bioinformatics and Computational Genomics Laboratory, Princess Margaret Cancer Center,University Health Network, Toronto, Ontario, Canada
    - &mbp Department of Medical Biophysics, University of Toronto, Toronto, Canada
  email: benjamin.haibe.kains@utoronto.ca
#package: PredictioR
output:
  BiocStyle::html_document
abstract: |
  Description of your vignette
vignette: |
  %\VignetteIndexEntry{SignatureSets: A Package For RNA Immunotherapy Signatures}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Introduction

The `SignatureSets` package provides access to a curated and extensive compendium of RNA-based Immuno-Oncology (IO) signatures. All signatures included in this package are published in peer-reviewed literature and publicly available through trusted repositories and resources. For more details, please refer to the manuscript [PMID 36055464](https://pubmed.ncbi.nlm.nih.gov/36055464/) or explore the PredictIO web application [predictio.ca](https://predictio.ca/) to easily mine this large compendium of published signatures.


```{r load, eval=TRUE, results='hide', message=FALSE, warning=FALSE}

library(data.table)
library(kableExtra)
library(summarytools)
library(stringr)

```


# Curated IO RNA-Based Signatures 

The SignatureSets package provides access to a curated and comprehensive compendium of 55 RNA-based IO signatures. Of these, 30 signatures (55%) are associated with sensitivity to IO therapy, while 25 signatures (45%) are linked to resistance mechanisms. 

```{r signature data}

app_dir <- str_split(rstudioapi::getActiveDocumentContext()$path,'SignatureSets')[[1]][1]
dir_GeneSig <- file.path(app_dir, 'SignatureSets', 'data')
dir_GeneSig_info <- file.path(app_dir, 'SignatureSets', 'data-raw')

signature <- read.csv( file.path(dir_GeneSig_info, "signature_information.csv")) 
signature$signature <- as.character( signature$signature )
signature$method <- as.character( signature$method )

kable(signature[1:5, c("signature", "method", "PMID", "information")], format="html", 
          table.attr = "style='width:80%;'", 
      row.names = FALSE) %>% 
      kableExtra::kable_styling()

```

The above table shows an overview of IO signatures data and methods to compute the signatures' scores including weighted mean, Gene Set Variation Analysis (GSVA) [PMID 23323831](https://pubmed.ncbi.nlm.nih.gov/23323831/), and specific computational algorithms. These signatures were extracted from the literature and manually annotated using GENCODE version 40 [PMID: 36420896](https://pubmed.ncbi.nlm.nih.gov/36420896/), with HUGO Gene Symbols as the primary identifiers linked to Entrez Gene IDs and Ensembl Gene IDs.

## Load Curated Immunotherapy Signature
     
As an example, ADO ([PMID 31953314](https://pubmed.ncbi.nlm.nih.gov/31953314/)) is a RNA signature capturing the adenosine (ADO) pathway activity. This adenosine signature was significantly associated with reduced efficacy of anti-PD1 therapy in published cohorts. To get the signature score, the GSVA approach is applied. 

```{r load signature ADO}

load(file.path(dir_GeneSig, "ADO_Sidders.rda"))
 
kable(sig, format="html", 
          table.attr = "style='width:30%;'", 
      row.names = FALSE) %>% 
      kableExtra::kable_styling()
                  
```

# Prognostic Landscape of the Curated Immunotherapy Signatures

Principal component analysis of the overlapping genes between each curated signature. Clusters were determined using Affinity Propagation Clustering. Kyoto Encyclopedia of Genes and Genomes #pathway enriched within each cluster was computed from enrichR.

```{r echo=FALSE, fig.height = 10, fig.width= 14}

knitr::include_graphics(file.path(app_dir, 'SignatureSets', 'utils', 'PatternSignatures.png'))

```

# References

Bareche, Y., Kelly, D., Abbas-Aghababazadeh, F., Nakano, M., Esfahani, P. N., Tkachuk, D., ... & Stagg, J. (2022). Leveraging big data of immune checkpoint blockade response identifies novel potential targets. Annals of Oncology, 33(12), 1304-1317.\

# SessionInfo

```{r sessionInfo, echo=FALSE}
#sessionInfo()
```

